### ISSUE TITLE
tidyclust integration notes

### ISSUE THREAD

**Issue by DavisVaughan** [tidymodels/workflows#176]
Rather than making the most minimal change possible, I think it will be better for the long term code health of workflows to add tidyclust as a new model "action" (this is what actions were intended for anyways). This allows you:

- To isolate the tidyclust changes (for the most part) in `fit-action-model-tidyclust.R`
- To add a new `add_model_tidyclust()` API that is simpler and separate from `add_model()` (i.e. no `formula` argument for model formulas)
- To use `check_conflicts()` to ensure that the user can't add both a tidyclust model and a parsnip model to the workflow
  - We would add a method for both `action_model` and `action_model_tidyclust` types

Rather than Importing or Suggesting tidyclust, since it is so heavy I think we should just use something like `dplyr:::import_vctrs()` (but not in `.onLoad()` because we dont know if tidyclust will be installed) to access functions from tidyclust that we need to call. Since we own both packages, I think that this is okay.

**Comment by simonpcouch** [comment]
I’ll go ahead and revert the `celery` branch to where it had been yesterday so that dev tidyclust can continue to rely on that branch for now. :)

**Comment by DavisVaughan** [comment]
Some thoughts about how to factor `new_action_model()` to handle unsupervised specs

```r
new_action_model <- function(spec, formula, ...) {
  if (is_model_spec(spec)) {
    new_action_model_spec(spec, formula)
  } else if (is_unsupervised_spec(spec)) {
    if (!is.null(formula)) {
      abort("cant use formulas here")
    }
    new_action_unsupervised_spec(spec)
  } else {
    abort("unknown spec")
  }
}
```

**Comment by simonpcouch** [comment]
Addressed in #184!
